python/plugins/processing/gui/TestTools.py

```diff
@@ -33,81 +36,153 @@
 from PyQt4.QtGui import QMessageBox, QDialog, QVBoxLayout, QTextEdit
 
 from processing.core.Processing import Processing
-from processing.core.outputs import OutputNumber
-from processing.core.outputs import OutputString
-from processing.core.outputs import OutputRaster
-from processing.core.outputs import OutputVector
-from processing.tools import vector, dataobjects
+from processing.core.outputs import (
+    OutputNumber,
+    OutputString,
+    OutputRaster,
+    OutputVector
+)
+
+from processing.core.parameters import (
+    ParameterRaster,
+    ParameterVector,
+    ParameterMultipleInput
+)
+
+
+def extractSchemaPath(filepath):
+    """
+    Trys to find where the file is relative to the QGIS source code directory.
+    If it is already placed in the processing or QGIS testdata directory it will
+    return an appropriate schema and relative filepath
+
+    Args:
+        filepath: The path of the file to examine
+
+    Returns:
+        A tuple (schema, relative_file_path) where the schema is 'qgs' or 'proc'
```

Comment:
I don't see where schema gets set to `'qgs'`. Seems like it's either `None` or `'proc'`.